### PR TITLE
[7.x] PHPUnit Followup

### PR DESCRIPTION
**Pull request by GrahamCampbell**
Goes with https://github.com/laravel/laravel/pull/5194.

### CODE REVIEW

File: src/Illuminate/Foundation/Testing/TestResponse.php [hunk 1]
```diff
@@ -4,9 +4,9 @@
 
 use ArrayAccess;
 use Closure;
+use GrahamCampbell\TestBenchCore\Assert as PHPUnit;
```
Comment (driesvints) [inline src/Illuminate/Foundation/Testing/TestResponse.php]: Doesn't this now make it a requirement?

Comment (GrahamCampbell) [reply src/Illuminate/Foundation/Testing/TestResponse.php]: I'd say no. We do the same with Mockery in loads of places, but we don't make it a requirement.

image

Comment (GrahamCampbell) [reply src/Illuminate/Foundation/Testing/TestResponse.php]: In fact, we do the same for all our optional dependencies. We use them, and then stuff breaks if they are not installed.

Comment (driesvints) [reply src/Illuminate/Foundation/Testing/TestResponse.php]: But surely the amount of usages of the `Assert` class will break things for people who forget to require the package?

Comment (GrahamCampbell) [reply src/Illuminate/Foundation/Testing/TestResponse.php]: Yeh. The same will happen if they forget to require mockery though?

Comment (driesvints) [reply src/Illuminate/Foundation/Testing/TestResponse.php]: Mockery is a dependency of laravel/laravel so it's unlikely that people don't have that.

Comment (GrahamCampbell) [reply src/Illuminate/Foundation/Testing/TestResponse.php]: Yeh, I'm planning on making testbench-core a dev-dependency of laravel/laravel. ;)

Comment (GrahamCampbell) [reply src/Illuminate/Foundation/Testing/TestResponse.php]: https://github.com/laravel/laravel/pull/5194

**Comment by taylorotwell** [comment]
Is all of this only to support the arraySubset feature? If so, can we just add that part of testbench-core into Laravel?

File: src/Illuminate/Foundation/Testing/TestResponse.php (hunk 1 above)
Comment (crynobone) [reply src/Illuminate/Foundation/Testing/TestResponse.php]: This changes is actually slightly different from other deps.

From my understanding the requirement for testbench-core was merely for array subset. However with this even `PHPUnit::assertTrue()` now also depends on testbench-core.

**Comment by taylorotwell** [comment]
So, explain again why this can't just be a requirement of the framework instead of forcing us to put it in `laravel/laravel`? What would it break?

**Comment by crynobone** [comment]
> If so, can we just add that part of testbench-core into Laravel?

With this exact changes, `orchestra/testbench-core` might also requires `graham-campbell/testbench-core` because some of the `TestResponse` related test that we have will definitely failed.

**Comment by GrahamCampbell** [comment]
> So, explain again why this can't just be a requirement of the framework instead of forcing us to put it in laravel/laravel? What would it break?

For the same reason we don't make phpunit or mockery a requirement of the framework. ;)

**Comment by crynobone** [comment]
I feel that it a bit overkill to have an additional package to mainly just to be used once in `TestResponse::assertJson()`. 

However I would see a potential benefit if we have a set of assertion that can be used by not only Laravel, Lumen, BrowserKit Testing and Dusk such as `assertJson()`, `assertJsonPath()`, `assertExactJson()`, `assertJsonFragment()`, `assetJsonMissing()`, `assertJsonMissingExact()`, `assertJsonStructure()`, `assertJsonCount()`. If we have that then that library would be very useful to us.

**Comment by driesvints** [comment]
I agree with @crynobone. Can we find a different solution for all this? I feel like relying on another package is just gonna overcomplicate things.

**Comment by GrahamCampbell** [comment]
I could bring my code into the framework?

**Comment by GrahamCampbell** [comment]
Happy to do that, and drop the dependency on all versions.

**Comment by driesvints** [comment]
> I could bring my code into the framework?

Yes that might be a good alternative. Can you send in a PR to the 6.x branch? I was going to look into that myself but would be cool if you could take that up. Thanks.

**Comment by GrahamCampbell** [comment]
Sure.

**Comment by GrahamCampbell** [comment]
We're going to move the original code into the laravel core on 6.x, mitigating the need for this, or any extra deps.

**Pull request closed without merging**